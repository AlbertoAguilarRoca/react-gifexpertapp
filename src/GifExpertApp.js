import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

//Componente principal
const GifExpertApp = ({defaultCategories = []}) => {

    //Hook con las categorias de Gif
    //const [categories, setCategories] = useState(['One Punch']);
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            {/* Componente para añadir categorias a través de un form*/}
            <AddCategory setCategories = {setCategories} />
            <hr/>

            <div>
                {
                    categories.map((category) => (
                        <GifGrid 
                            key={category}
                            category={category} 
                        />
                    ))
                }
            </div>
        </>
    );
}

export default GifExpertApp;