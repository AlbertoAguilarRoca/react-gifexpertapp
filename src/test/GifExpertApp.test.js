import React from 'react';
import { shallow } from 'enzyme';

import GifExpertApp from '../GifExpertApp';

describe('pruebas en <GifExpertApp />', () => {

    test('debe de mostrar correctamente <GifExpertApp />', () => {

        const wrapper = shallow(<GifExpertApp />)

        expect(wrapper).toMatchSnapshot();

    });

    test('debe de mostrar una lista de categorias', () => {
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories = {categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    });

});