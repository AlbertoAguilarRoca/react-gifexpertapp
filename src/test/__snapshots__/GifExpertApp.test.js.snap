// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExpertApp /> debe de mostrar correctamente <GifExpertApp /> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div />
</Fragment>
`;

exports[`pruebas en <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </div>
</Fragment>
`;
